code/src/main/java/com/onlinecontacttracing/messaging/MessageServlet.java

```diff
@@ -0,0 +1,97 @@
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.util.Collections;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Projection;
+
+@WebServlet("/messaging")
+public class MessageServlet extends HttpServlet {
+  private SystemMessage systemMessage;
+  private LocalityResource localityResource;
+  private CustomizableMessage customizableMessage;
+  private String userMessage;
+  private PositiveUser user;
+  //for backend use
+  private String compiledMessage;
+  //for frontend display
+  private List<String> statusMessage = new ArrayList<String> ();
+
+
+  public MessageServlet(SystemMessage systemMessage, LocalityResource localityResource, CustomizableMessage customizableMessage, PositiveUser user) {
+    this.systemMessage = systemMessage;
+    this.localityResource = localityResource;
+    this.customizableMessage =  customizableMessage;
+    this.userMessage = customizableMessage.getMessage();
+    this.user = user;
+
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html;");
+    String messageLanguage = request.getParameter("language");
+    response.getWriter().println(statusListToShowUser(messageLanguage));
+  }
+
+  public void checkForFlags() {
+    CheckMessagesForFlags flagChecker = new CheckMessagesForFlags();
+    statusMessage = CheckMessagesForFlags.findTriggeredFlags(flagChecker, user, userMessage);
+  }
+
+  //if flags are triggered, returns message and list of errors, else returns message
+  public List<String> statusListToShowUser(String messageLanguage) {
+    String translatedResourceMessage;
+    String translatedSystemMessage;
+    user.incrementAttemptedEmailDrafts();
+    checkForFlags();
+    
+    if (messageLanguage.equals("SP")) {
+    translatedResourceMessage = localityResource.getEnglishTranslation();
+    translatedSystemMessage = systemMessage.getEnglishTranslation();
+    }
+    else {
+      translatedResourceMessage = localityResource.getEnglishTranslation();
+      translatedSystemMessage = systemMessage.getEnglishTranslation();
+    }
+    
+    //if any flags are triggered, do not want to include the user's message in the message sent to the backend
+    if (statusMessage.size() != 0){
+      compiledMessage = translatedSystemMessage.concat(translatedResourceMessage);
+    } else {
+      compiledMessage = (translatedSystemMessage.concat(userMessage).concat(translatedResourceMessage);
+    }
+    statusMessage.add(translatedSystemMessage.concat(userMessage).concat(translatedResourceMessage));
+    return statusMessage;
+  
+  }
+
+  public String compileMessage () {
+    //returns message with no userMessage if there are errors, else return message with userMessage
+    if (statusMessages.size() > 1) {
```

Comment:
Is this an attempt at caching?

If so, don't do this. Just general construct the message each time. Or else, fully cache the message and get rid of this weird choice logic.